sklearn/model_selection/_split.py

```diff
@@ -610,7 +610,7 @@ def _iter_test_indices(self, X, y, groups):
         n_samples_per_group = np.bincount(groups)
 
         # Distribute the most frequent groups first
-        indices = np.argsort(n_samples_per_group)[::-1]
+        indices = np.argsort(n_samples_per_group, kind="stable")[::-1]
```

Comment:
Technically once we support numpy v2.0+ only, we could set this as `stable=True`.

https://numpy.org/doc/stable/reference/generated/numpy.argsort.html

Perhaps we can add an in-line comment:

```
# TODO: replace with np.argsort(n_samples_per_group, stable=True)[::-1] when numpy v2.0 is the minimum supported version
```

Reply:
Sure, though, the `stable=True` docs on numpy 2.0 mention it just sets `kind="stable"` behind the scenes.

Reply:
Oh fair point. I missed that. Nvm then on this comment!